controller/controller_server/sync_lock.py

```diff
@@ -0,0 +1,48 @@
+import threading
+from collections import defaultdict
+
+from decorator import decorator
+
+from controller.controller_server.errors import VolumeAlreadyProcessingError
+
+from controller.common.csi_logger import get_stdout_logger
+
+logger = get_stdout_logger()
+
+
+class SyncLock:
+    def __init__(self):
+        self._lock = threading.Lock()
+        self._lock_ids = defaultdict(set)
+
+    def add_volume_lock(self, lock_key, lock_id, action_name):
+        logger.debug(("Lock for action: {}, Try to acquire lock for volume: {}".format(action_name, lock_id)))

```

Comment:
volume?
please include the `lock_key` in the messages

Reply:
done